### PR TITLE
tiles: rewrite mux to stem

### PR DESCRIPTION
**Pull request by mmcgee-jump**

### CODE REVIEW

**Review by ripatel-fd — CHANGES_REQUESTED** [review]
Large refactors like these should come with unit tests. 
fd_stem doesn't appear to have any test coverage

**Comment by mmcgee-jump** [comment]
That's coming in a separate change, this is getting too big.

**Comment by ripatel-fd** [comment]
> That's coming in a separate change, this is getting too big.

Changing test_mux to test_stem instead of deleting it might reduce the diff

**Comment by mmcgee-jump** [comment]
That test is a bit limited, and it also didn't run in CI.

**Comment by mmcgee-jump** [comment]
todo: switch the printer for counter/gauge to use converter as well

File: src/disco/stem/fd_stem.c [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+  }
+
+  /* out frag stream init */
+
+  cr_avail = 0UL;
+
+  out_depth  = (ulong *)FD_SCRATCH_ALLOC_APPEND( l, alignof(ulong), out_cnt*sizeof(ulong) );
+  out_seq    = (ulong *)FD_SCRATCH_ALLOC_APPEND( l, alignof(ulong), out_cnt*sizeof(ulong) );
+
+  ulong cr_min = fd_ulong_if( !out_cnt, 128UL, ULONG_MAX );
```
Comment (ptaffet-jump) [inline src/disco/stem/fd_stem.c:267]: Maybe call this `depth_min`

Comment (ptaffet-jump) [reply src/disco/stem/fd_stem.c:267]: And then keep the variable formerly called `cr_max`. It's just confusing to see a variable change names from `cr_max` to `cr_min` without changing roles

**Review by ptaffet-jump — APPROVED** [review]
I looked at `fd_stem` a little, and the rest of the PR almost not at all, but I'm stamping to unblock

**Review by lheeger-jump — APPROVED** [review]

**Review by ripatel-fd — DISMISSED** [review]

**Comment by ripatel-fd** [comment]
Kicking CI to rerun failed jobs. (flaky CI environment)

**Review by ripatel-fd — APPROVED** [review]

**Review by ptaffet-jump — APPROVED** [review]
Restamping

**Pull request merged**